tests/setup.py

```diff
@@ -147,3 +147,15 @@ def grant_permission_to_group(self, permission, argument, group):
             permission_id=permission_obj.id, group_id=group_obj.id, argument=argument
         )
         grant.add(self.session)
+
+    def create_group_request(self, user, group, role="member"):
```

Comment:
This doesn't do what the function name says so far as I can tell. It just adds the user directly to the group. (Maybe it creates a request as a side effect, but I think you instead want to create a request explicitly and directly.)

Reply:
Ah, I see. Yeah, this looks good.